### PR TITLE
Fix bug 1527507 - Move all constraints into a constraints file.

### PR DESCRIPTION
**Pull request by jotes**
Hey @adngdb 

There's still a few things to do before the  merge (look below). However, I would love to know your opinion if that PR has the potential to be merged ;D
I think you should check following things:
* if the names of files are correct (or should we move them into a separate directory, perhaps `requirements`)
* if the proposed structure of the constraints file is okay (basically it's not different from the previous proposal).
* Any other thing you have on your mind

Thanks in advance :-)

TODO:
* [x] fix the docs
* [x] pass `-c` flag in  all pip calls
* [x] double check if all constraints are intact?
* [x] update the commit message

### CODE REVIEW

**Comment by adngdb** [comment]
I like this very very much! Combining those two approaches makes both files very easy to understand and maintain, so yeah, a big :+1:.

On the topic of folders and file names, I'd move all our requirements files into a `requirements` folder and rename them `default.txt`, `dev.txt`, `test.txt` and `constraints.txt`. 

Thanks @jotes!

**Comment by adngdb** [comment]
@jotes note that I just made some changes to our documentation, so you might want to wait for them to be merged and rebase your branch before updating the doc on your side.

https://github.com/mozilla/pontoon/pull/1211

**Comment by jotes** [comment]
@adngdb thank for the update :-)

**Comment by jotes** [comment]
@adngdb could you review this PR? Thanks in advance :-)

**Comment by jotes** [comment]
Howdy,
@adngdb I removed all duplicated `-c` parameters from the PR.

@Pike 
Thanks to your advice, I was able to generate a reverse dependency tree and strip it to the second level.
You can find the result here: https://gist.github.com/jotes/42bad6c81082173a149286d63497af16
There are some libraries required by multiple libraries (e.g. six). 
I think I can suggest the new format/convention mixed with your approach https://github.com/mozilla/elmo/blob/master/requirements/constraints.txt
@adngdb is it okay for you?

File: requirements/dev.txt [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+# Dependencies that do not come from pypi (eg. eggs from github) are listed at
+# the end of that list.
+#
+# The second group are should be added to requirements-constraints.txt and follow conventions
+# explained there.
+# ----------------------------------------------------------------------------------
+#
+# requirements/default.txt contains all dependencies required to run python in the production environment.
+-r default.txt
+-c constraints.txt
```
Comment (Pike) [inline requirements/dev.txt]: The `-c` is covered by `default.txt`, I think

Comment (jotes) [reply requirements/dev.txt]: Fixed, thanks :)

**Comment by jotes** [comment]
[Update]
I implemented the last comment from @Pike: 
https://github.com/mozilla/pontoon/pull/1204/commits/2354733adf9992f64f76cf9cbbcc62097e086342 

I'm not sure if that's a blocker, but I reformatted constraints.txt (inspired by the constraints file from Elmo):
https://github.com/mozilla/pontoon/pull/1204/commits/70a6a8f7c471ea985b50b663643cefd6ed9f55ed

@Pike could you review too? Thanks in advance (and I apologize for all late-night errors :().
@adngdb If the scope of changes added in this PR is getting to big,  I can remove the last commit.

File: requirements/dev.txt [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+# are used by the application directly) are sorted by alphabetical order.
+# Dependencies that do not come from pypi (eg. eggs from github) are listed at
+# the end of that list.
+#
+# The second group are should be added to requirements-constraints.txt and follow conventions
+# explained there.
+# ----------------------------------------------------------------------------------
+#
+# requirements/default.txt contains all dependencies required to run python in the production environment.
+-r default.txt
```
[the remark this answers is not in the document]
Comment (adngdb) [reply requirements/dev.txt]: Ah, but that's not new, is it? It was already like this before? Hmmm really I don't know which way is better, so you decide! :)

File: CONTRIBUTING.rst [hunk 3]
```diff
@@ -147,23 +147,23 @@ Code reviews should review the changes in the context of the rest of the system.
 Dependencies
 ============
 
-Dependencies for production Pontoon are in ``requirements.txt``. Development dependencies are in
-``requirements-dev.txt``. They need to be pinned and hashed, and we use `hashin <https://pypi.python.org/pypi/hashin>`_ for that.
+Dependencies for production Pontoon are in ``requirements/default.txt``. Development dependencies are in
+``requirements/dev.txt``. They need to be pinned and hashed, and we use `hashin <https://pypi.python.org/pypi/hashin>`_ for that.
 
 Note that we use a specific format for our dependencies, in order to make them more maintainable. When adding a new requirement, you should add it to the appropriate section and group it with its sub-dependencies if applicable.
```
[the remark this answers is not in the document]
Comment (jotes) [reply CONTRIBUTING.rst]: Thanks, updated.

File: requirements/dev.txt (hunk 2 above)
[the remark this answers is not in the document]
Comment (jotes) [reply requirements/dev.txt]: I would leave it as it is for the time being and create a follow-up bug later (if that's okay with you).

**Comment by adngdb** [comment]
@jotes something went wrong with the history of this branch, there are commits that shoudn't be there. Can you take a look? I guess you'll need to `git rebase -i HEAD~8` and remove the few commits that don't belong, then `git rebase master` to make sure you're up-to-date with master.

**Comment by adngdb** [comment]
Damn, sorry, there's a conflict now... My apologies for not coming back here sooner, I missed that you did the update. 

Side-note, when you force-push to a branch, it doesn't send a notification, so you might want to ping in the PR in such cases.

**Comment by jotes** [comment]
@adngdb 
It is rebased right now :D

**Comment by adngdb** [comment]
Great, docker build and tests passing locally, this is excellent! Thank you very much for your work @jotes!

**Pull request merged**